src/config/env.rs

```diff
@@ -0,0 +1,114 @@
+//! Configuration from environment variables.
+
+use std::fmt;
+
+use camino::{Utf8Path, Utf8PathBuf};
+
+use super::{Config, LogLevel, SettingSource};
+
+//----------- EnvSpec ----------------------------------------------------------
+
+/// Configuration-related environment variables.
+#[derive(Clone, Debug)]
+pub struct EnvSpec {
+    /// The configuration file to load.
+    pub config: Option<Box<Utf8Path>>,
+
+    /// The minimum severity of messages to log.
+    pub log_level: Option<LogLevel>,
+
+    /// The file to write logs to.
+    pub log_file: Option<Box<Utf8Path>>,
+}
+
+impl EnvSpec {
+    /// Process environment variables.
+    pub fn process() -> Result<Self, EnvError> {
+        Ok(Self {
+            config: match std::env::var("NAMESHED_CONFIG") {
```

Comment:
Uou use _FILE below in NAMESHED_LOG_FILE so perhaps this should also have a _FILE suffix, or both should be called _PATH?

Reply:
I think I slightly prefer PATH as FILE is never clear to me if it includes the path to the containing directory or if that is defined elsewhere.